src/proxy/outbound.rs

```diff
@@ -195,19 +198,89 @@ impl OutboundConnection {
             metrics,
         ));
 
-        let res = match req.protocol {
-            Protocol::HBONE => {
+        let res = match (
+            req.protocol,
+            req.actual_destination_workload
+                .as_ref()
+                .and_then(|wl| wl.network_gateway.as_ref()),
+        ) {
+            (_, Some(_)) => {
```

Comment:
I don't think there is a guarantee that something that has a network_gateway supports HBONE. Or that something that has a network_gateway is even in a different network

Reply:
So we should check that the protocol is HBONE to do double hbone? what about the inner tunnel? I guess my question is how should we explicitly encode protocol information.? Maybe have it be a field in `network_gateway`?

Reply:
@howardjohn does WDS/network gateway need a new `protocol` field? or are we good just checking that the protocol is HBONE and returning an Error in the rest of the cases? cc @keithmattix

Reply:
Sorry github hid this comment so end up saying something similar but with slightly more context in https://github.com/istio/ztunnel/pull/1429#discussion_r2008209563.

I think the WDS protocol has enough information, we just need to use the context of (src,dst) to determine whether a cross network traversal is required

Reply:
So just looking at the source/ dest networks should give enough info on whether to use the network gateway , and I can put this logic next to the waypoint logic in ‘build_request’ (I think that’s what the function is called).